src/xss-common-qsort.h

```diff
@@ -838,6 +894,26 @@ X86_SIMD_SORT_INLINE void avx512_qsort(T *arr, arrsize_t arrsize)
     }
 }
 
+template <typename T>
+void avx2_qsort(T *arr, arrsize_t arrsize)
+{
+    using vtype = xss::avx2::ymm_vector<T>;
+    if (arrsize > 1) {
+        /* std::is_floating_point_v<_Float16> == False, unless c++-23*/
+        if constexpr (std::is_floating_point_v<T>) {
+            arrsize_t nan_count
+                    = xss::avx2::replace_nan_with_inf(arr, arrsize);
```

Comment:
We should be able to reuse the template function defined in this file `replace_nan_with_inf(T *arr, arrsize_t size)` and not re-write this function.